lib/compilers/typescript-native.ts

```diff
@@ -45,6 +46,7 @@ export class TypeScriptNativeCompiler extends BaseCompiler {
 
         this.tscJit = this.compilerProps<string>(`compiler.${this.compiler.id}.exe`);
         this.tscSharedLib = this.compilerProps<string>(`compiler.${this.compiler.id}.sharedlibs`);
+        this.tscVersion = /tsc-\d+\.\d+-pre-alpha(\d+)/g.exec(this.tscJit)?.[1] || "0";
```

Comment:
Check how https://github.com/compiler-explorer/compiler-explorer/blob/main/lib/compilers/zig.ts handles differing behaviour per version in a "safer" way :)

Reply:
thx, updated